plutus-core/testlib/PlutusCore/Generators/PIR/GenTm.hs

```diff
@@ -0,0 +1,317 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module PlutusCore.Generators.PIR.GenTm
+  ( module PlutusCore.Generators.PIR.GenTm
+  , module Export
+  , Arbitrary (..)
+  , Gen
+  ) where
+
+import Control.Monad.Except
+import Control.Monad.Reader
+
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.Map.Strict (Map)
+import Data.Map.Strict qualified as Map
+import Data.Maybe
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Data.Set.Lens (setOf)
+import Data.String
+import Test.QuickCheck (Arbitrary (..), Gen)
+import Test.QuickCheck qualified as QC
+import Test.QuickCheck.GenT as Export hiding (var)
+
+import PlutusCore.Default
+import PlutusCore.Name
+import PlutusIR
+import PlutusIR.Compiler
+import PlutusIR.Subst
+
+import PlutusCore.Generators.PIR.Common
+import PlutusCore.Generators.PIR.Utils
+
+instance MonadReader r m => MonadReader r (GenT m) where
```

Comment:
upstream?

Reply:
I think they intentionally don't depend on `mtl` in `quickcheck-transformer`? It's hard to imagine they forgot all such instances by accident. If anything, we should just create a new package with these instances. I checked, it's [not](https://packdeps.haskellers.com/reverse/quickcheck-transformer) on hackage.

Reply:
... maybe just forget about it :sweat_smile: also upstream hasn't updated for years and is in darcs :grimacing: